tests/ui/array-slice-vec/slice-panic-1.rs

```diff
@@ -5,15 +5,17 @@
 
 // Test that if a slicing expr[..] fails, the correct cleanups happen.
 
-
 use std::thread;
+use std::sync::atomic::{AtomicUsize, Ordering};
 
 struct Foo;
 
-static mut DTOR_COUNT: isize = 0;
+static DTOR_COUNT: AtomicUsize = AtomicUsize::new(0);
 
 impl Drop for Foo {
-    fn drop(&mut self) { unsafe { DTOR_COUNT += 1; } }
+    fn drop(&mut self) {
+        DTOR_COUNT.fetch_add(1, Ordering::SeqCst);
```

Comment:
Here and elsewhere, it'd be better to use `Ordering::Relaxed`.  When we're just using these to atomically increment counters (and not to e.g. implement some locking primitive), `Relaxed` has all of the necessary guarantees.